internal/app/init/internal/platform/baremetal/baremetal.go

```diff
@@ -86,7 +86,7 @@ func (b *BareMetal) UserData() (data *userdata.UserData, err error) {
 		return data, nil
 	}
 
-	return userdata.Download(option)
+	return userdata.Download(option, nil)
```

Comment:
Throwing nil vals in for headers on all the platform types that aren't gcloud. May be possible to do this better later on.